discord/message_handlers.go

```diff
@@ -45,22 +46,32 @@ func (bot *Bot) handleNewGameMessage(guild *GuildState, s *discordgo.Session, m
 	bot.LinkCodeLock.Unlock()
 
 	var hyperlink string
-	var hyperlinkPort string
-	if bot.extPort != "" {
-		if bot.extPort == "protocol" {
-			hyperlinkPort = ""
-		} else {
-			hyperlinkPort = ":" + bot.extPort
+	urlregex := regexp.MustCompile(`http(?P<secure>s?)://(?P<host>[^:]*):?(?P<port>.*)`)
```

Comment:
I recommend tightening up this expression to avoid some unexpected matching. (For example, if authentication is required in the connect URL.)

Try: `^http(?P<secure>s?):\/\/(?P<host>[\w.-]+)(?::(?P<port>\d+))?$`
See https://regex101.com/r/tu2aSG/1

Reply:
Just tested if the `[\w.-]` would break punycode domains but those don't seem to work anyway

Reply:
Maybe, but I screwed up there anyway. It should support `@` and `:` as well.

I _think_ the full set of characters allowed there should be `[\w@:.-]+?`. (Lazy, so it doesn't gobble the port.

Are there any other characters I'm missing? Would it just be safer to do `[^/]`?

Reply:
Updated link: https://regex101.com/r/tu2aSG/2